Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -372,8 +562,13 @@ function Enable-MicrosoftDefender
                         $nonCompliantMDCTierResourcetype += $resource  
                     }
 
-                    if ($EnableServers -eq $true -and $_.Name -eq "VirtualMachines" -and  $_.PricingTier -ne $reqMDCTier) {
-                        $nonCompliantMDCTierResourcetype += $resource 
+                    if ($EnableServers -eq $true -and $_.Name -eq "VirtualMachines") {
+                        $resource = $_ | Select-Object Name, PricingTier, Id, SubPlan, Extensions
+                        $complianceResult = Check-VirtualMachineCompliance -resource $resource -subscriptionId $subscriptionId -reqMDCTier $reqMDCTier
+
+                        if ($complianceResult) {
+                            $nonCompliantMDCTierResourcetype += $complianceResult
```

Comment:
if condition and the assignment inside are not in sync

Reply:
+1

Reply:
Function is ultimately getting non compliant resources. So updated function name to Get-NonCompliantVirtualMachine.
and if statement to
if ($nonCompliant) {
                            $nonCompliantMDCTierResourcetype += $nonCompliant
                        }